### DISCUSSION TITLE
Question about computing the default value of an array for validation

### DISCUSSION THREAD

**Discussion by msarauer** [rjsf-team/react-jsonschema-form#3259] [Q&A]
When computeDefaults is run with a schema of `"type": "array"`, with `"minItems": 1`, and an item of `"type":"string"` and no default specified in the schema, it is returning a default of `[undefined]`. If the default is then used in validation because I left the input empty, this fails the validation, because it is looking for an array containing a string. I feel like the default form state should be returning `undefined` not in an array because I don't want this failing my validation when I don't enter a value, unless this property is in the required array.

**Comment by anthonycaron** [comment]
I totally agree with this ! I actually just posted [the same comment](https://github.com/rjsf-team/react-jsonschema-form/issues/3602#issuecomment-1541487059) on an issue that's just been closed.

**Reply by anthonycaron** [reply]
I will try now that this issue has been fixed if the test I wrote passes.

**Reply by anthonycaron** [reply]
It does not pass. I will open an issue.

**Comment by anthonycaron** [comment]
Actually I'm not sure anymore. I'm thinking maybe the default form state should be an empty array. Because when we add `minItems` on an array it's a way to say this field is required. If we initialize it with `[]` then when we try to validate it, `ajv` will spit an error saying we should have a least `${minItems}` elements for this array, which is good I think.

**Comment by anthonycaron** [comment]
@heath-freenome Could you give us your opinion on this please ?

**Reply by heath-freenome** [reply]
Putting in undefined into the array should, if the UI is working correctly, cause an input to show up that should then have data provided for it. The new feature was added to only force this situation when the input is required. When it is not required, then the entire array missing min items will not be added and, hopefully, if no items have yet been entered, then the array will not be provided at all.

**Reply by anthonycaron** [reply]
So you mean that if the array is not required, then it shouldn't even appear in the data by default ?

**Reply by heath-freenome** [reply]
Yes, exactly. Hopefully that is acceptable.

**Reply by anthonycaron** [reply]
Yes it is for me :)

**Reply by anthonycaron** [reply]
But right now it's not working like this as shown in [the issue I opened](https://github.com/rjsf-team/react-jsonschema-form/issues/3656#issue-1703351764).
